provider/vault/vault.go

```diff
@@ -0,0 +1,174 @@
+	"github.com/bank-vaults/vault-sdk/vault"
+
+	"github.com/bank-vaults/secret-init/common"
+	"github.com/bank-vaults/secret-init/provider"
+)
+
+const ProviderName = "vault"
+
+type Provider struct {
+	isLogin        bool
+	client         *vault.Client
+	injectorConfig injector.Config
+	secretRenewer  injector.SecretRenewer
+	fromPath       string
+	revokeToken    bool
+}
+
+type sanitized struct {
+	secrets []provider.Secret
+	login   bool
+}
+
+// VAULT_* variables are not populated into this list if this is not a login scenario.
+func (s *sanitized) append(key string, value string) {
+	envType, ok := sanitizeEnvmap[key]
+	// If the key being appended is not present in sanitizeEnvmap, it signifies that
+	// it is not a VAULT_* variable.
+	// Additionally, in a login scenario, we include VAULT_* variables in the secrets list.
+	if !ok || (s.login && envType.login) {
+		// Path here is actually the secret's key
+		// (the environment variable name)
+		secret := provider.Secret{
+			Path:  key,
+			Value: value,
+		}
+
+		s.secrets = append(s.secrets, secret)
+	}
+}
+
+// passing daemonMode is a dirty way for now to avoid dependency
+func NewProvider(config *Config) (provider.Provider, error) {
+	clientOptions := []vault.ClientOption{vault.ClientLogger(clientLogger{slog.Default()})}
+	if config.TokenFile != "" {
+		clientOptions = append(clientOptions, vault.ClientToken(config.Token))
+	} else {
+		// use role/path based authentication
+		clientOptions = append(clientOptions,
+			vault.ClientRole(config.Role),
+			vault.ClientAuthPath(config.AuthPath),
+			vault.ClientAuthMethod(config.AuthMethod),
+		)
+	}
+
+	client, err := vault.NewClientWithOptions(clientOptions...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create vault client: %w", err)
+	}
+
+	// Accessing the application configuration is necessary
+	// to determine whether daemon mode is enabled
+	appConfig, err := common.LoadConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to load application config: %w", err)
+	}
+
+	injectorConfig := injector.Config{
+		TransitKeyID:         config.TransitKeyID,
+		TransitPath:          config.TransitPath,
+		TransitBatchSize:     config.TransitBatchSize,
+		IgnoreMissingSecrets: config.IgnoreMissingSecrets,
+		DaemonMode:           appConfig.Daemon,
+	}
+
+	var secretRenewer injector.SecretRenewer
+
+	if appConfig.Daemon {
+		sigs := make(chan os.Signal, 1)
+		signal.Notify(sigs)
+
+		secretRenewer = daemonSecretRenewer{client: client, sigs: sigs}
+		slog.Info("Daemon mode enabled. Will renew secrets in the background.")
+	}
+
+	return &Provider{
+		isLogin:        config.IsLogin,
+		client:         client,
+		injectorConfig: injectorConfig,
+		secretRenewer:  secretRenewer,
+		fromPath:       config.FromPath,
+		revokeToken:    config.RevokeToken,
+	}, nil
+}
+
+// LoadSecret's path formatting: <key>=<path>
+// This formatting is necessary because the injector expects a map of key:value pairs.
```

Comment:
would be good to also add an example of expected payload (similar to how its handled in EnvStore tests)
```suggestion
// This formatting is necessary because the injector expects a map of key=value pairs.
```